mongo/errors.go

```diff
@@ -449,15 +473,27 @@ func (mwe WriteException) Error() string {
 
 // HasErrorCode returns true if the error has the specified code.
 func (mwe WriteException) HasErrorCode(code int) bool {
-	if mwe.WriteConcernError != nil && mwe.WriteConcernError.Code == code {
-		return true
+	return hasErrorCode(mwe, code)
+}
+
+// ErrorCodes returns a list of error codes returned by the server.
+func (mwe WriteException) ErrorCodes() []int {
+	errorCodeSet := make(map[int]struct{})
+	for _, writeError := range mwe.WriteErrors {
+		errorCodeSet[writeError.Code] = struct{}{}
 	}
-	for _, we := range mwe.WriteErrors {
-		if we.Code == code {
-			return true
-		}
+
+	if mwe.WriteConcernError != nil {
+		errorCodeSet[mwe.WriteConcernError.Code] = struct{}{}
 	}
-	return false
+
+	// Deduplicate error codes.
+	errorCodes := make([]int, 0, len(errorCodeSet))
+	for code := range errorCodeSet {
+		errorCodes = append(errorCodes, code)
+	}
```

Comment:
Do we need to deduplicate error codes? If the server returns the same error for multiple ops, it might be useful to see multiple of the same error codes. Users who want a deduplicated view of errors can do that logic themselves.